mslib/mscolab/file_manager.py

```diff
@@ -250,6 +256,59 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def upload_file(self, file, upload_folder, file_token=None, subfolder=None, identifier=None):
+        """
+        Generic function to save files securely in specified directory with unique filename
+        and return the relative file path.
+        """
+        with fs.open_fs(upload_folder):
+            file_dir = fs.path.join(upload_folder, str(subfolder) if subfolder else "")
+            if sys.platform.startswith('win'):
+                file_dir = file_dir.replace('\\', '/')
+            if not os.path.exists(file_dir):
+                os.makedirs(file_dir)
```

Comment:
This should fail on windows or at least create a wrong directory, because `os.makedirs` expects the OS-native separator, which was just replaced by "/".

Three options:
1. Fully use the `fs` module:
    ```python
    with fs.open_fs(upload_folder) as _fs:
        if sys.platform.startswith('win'):
            file_dir = file_dir.replace('\\', '/')
        _fs.makedirs(str(subfolder) if subfolder else "", recreate=True)
    ```
2. Fully use the `os` module:
    ```python
    file_dir = os.path.join(upload_folder, str(subfolder) if subfolder else "")
    os.makedirs(file_dir, exist_ok=True)
    ```
3. Use the more modern `pathlib` module:
    ```python
    file_dir = pathlib.Path(upload_folder) / (subfolder if subfolder is not None else "")
    file_dir.mkdir(parents=True, exist_ok=True)
    ```

Mixing `fs` and `os`/`pathlib` always leads to issues.

Which option is appropriate depends on if we need the fs-related advantages here. I am not sure if we do, I don't think so really; my preference would be `pathlib`. @ReimarBauer what is your opinion?

Reply:
I've not looked up more on this yet:
https://github.com/Open-MSS/MSS/issues/2103

When we do this we will do a complete refactoring in hopefully one shot including docs and examples.

At current state because this is also not in the students proposal work we should keep fs.